rmgpy/rmg/main.py

```diff
@@ -804,6 +807,41 @@ def execute(self, **kwargs):
         
         self.finish()
     
+    def check_model(self):
+        """
+        Run checks on the RMG model
+        """
+        #check that no two species in core or edge are isomorphic
+        for i,spc in enumerate(self.reactionModel.core.species):
+            for j in xrange(i):
+                spc2 = self.reactionModel.core.species[j]
+                if spc.isIsomorphic(spc2):
+                    raise CoreError('Although the model has completed, species {0} is isomorphic to species {1} in the core.  Please open an issue on GitHub with the following output: {2}\n{3}'.format(spc.label,spc2.label,spc.toAdjacencyList(),spc2.toAdjacencyList()))
+                    
+        
+        for i,spc in enumerate(self.reactionModel.edge.species):
+            for j in xrange(i):
+                spc2 = self.reactionModel.edge.species[j]
+                if spc.isIsomorphic(spc2):
+                    logging.warning('species {0} is isomorphic to species {1} in the edge.  Note this does not affect the generated model.  If you would like to report this to help make RMG better please open a GitHub issue with the following output: {2}\n{3}'.format(spc.label,spc2.label,spc.toAdjacencyList(),spc2.toAdjacencyList()))
+        
+        for i,rxn1 in enumerate(self.reactionModel.core.reactions):
+            for j in xrange(i):
+                rxn2 = self.reactionModel.core.reactions[j]
+                if not(rxn1.duplicate and rxn2.duplicate) and rxn1.isIsomorphic(rxn2):
+                    logging.warning('rxn {0} in the core has duplicate reactions that are unmarked these reactions may duplicate each other:')
+                    logging.warning('index: {0}, \n rxn: {1}'.format(i,repr(rxn1)))
+                    logging.warning('index: {0}, \n rxn: {1}'.format(j,repr(rxn2)))
+                    
+        for i,rxn1 in enumerate(self.reactionModel.edge.reactions):
+            for j in xrange(i):
+                rxn2 = self.reactionModel.edge.reactions[j]
+                if not(rxn1.duplicate and rxn2.duplicate) and rxn1.isIsomorphic(rxn2):
+                    logging.warning('rxn {0} in the edge has duplicate reactions that are unmarked these reactions may duplicate each other:')
+                    logging.warning('index: {0}, \n rxn: {1}'.format(i,repr(rxn1)))
+                    logging.warning('index: {0}, \n rxn: {1}'.format(j,repr(rxn2)))
```

Comment:
You can use `{1!r}` instead of calling `repr` on the reactions.

Reply:
fixed